org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinkingResource.java

```diff
@@ -95,6 +103,12 @@ private CyclicLinkingException(Object detailMessage, Triple<EObject,EReference,I
 
 	private boolean eagerLinking = false;
 
+	@Named(CYCLIC_LINKING_DECTECTION_COUNTER_LIMIT)
+	@Inject(optional=true)
+	private int cyclicLinkingDectectionCounterLimit = 100;
```

Comment:
I'd make these protected sind `resolving` is protected, too.

Reply:
done